src/ex2/ItemManager.js

```diff
@@ -0,0 +1,170 @@
+import UIManager from "./UIManager.js";
+import Item from "./ItemClass.js";
+import PokemonClient from "./PokemonClient.js";
+
+class ItemManager {
+  constructor() {
+    this.itemIdCounter = 2000;
+    this.pokemonClient = new PokemonClient();
+    this.uiManager = new UIManager();
+    this.fethcedPokemons = [];
+    this.pokemonErrorBatch = [];
+    this.itemList = [];
+    this.allPokemonsJson = null;
+    this.clearAllBtn = document.getElementById("clearAllBtnId");
+    // this.trashButton = null;
+    this.clearAllBtn.addEventListener("click", () => {
+      this.handleClearAllTodos();
+    });
+  }
+
+  /**
+   *
+   * @param {string} inputValue
+   * @returns
+   */
+  parseInputValue(inputValue) {
+    inputValue = inputValue.trim();
+    if (!inputValue) {
+      return alert("Error, the Todo input is empty");
+    }
+
+    let promises = [];
+
+    for (const item of inputValue.split(/\s*,\s*/)) {
+      if (/^\d+$/.test(item)) {
+        const pokemonNumber = Number(item);
+
+        const pokemonInList = this.fethcedPokemons.find(function (
+          pokemon,
+          index
+        ) {
+          if (pokemon.id == pokemonNumber) {
+            return true;
+          }
+        });
+        let promise = null;
+        if (!pokemonInList) {
+          promise = this.fetchPokemonAndAddToList(pokemonNumber);
+        }
+        promises.push(promise);
```

Comment:
why do we want to push "null" to the promises? in the case when we already fetched it. I think you should push the promise only if you have one.